README.md

```diff
@@ -1,63 +1,73 @@
+| `SECURITY.md`     | Security disclosures.                      |
+| `tsconfig.*.json` | TypeScript configuration files.            |
+| `tslint.json`     | TS Lint configuration files.               |
 
 ## Prerequisites
 
-Outline the required components and tools that a user might need to have on their machine in order to run the sample. This can be anything from frameworks, SDKs, OS versions or IDE releases.
+[Node](https://nodejs.org/en/) must be installed to run this sample.
 
 ## Setup
 
-Explain how to prepare the sample once the user clones or downloads the repository. The section should outline every step necessary to install dependencies and set up any settings (for example, API keys and output folders).
+1. Register a new application in the [Azure Portal](https://portal.azure.com). Ensure that the application is [enabled for the implicit flow](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration).
+2. Open the [src/app/app.module.ts](./src/app/app.module.ts) file and provide the required configuration values (inside `MsalModule.forRoot`).
+3. On the command line, navigate to the root of the repository, and run `npm install` to install the project dependencies via npm.
 
 ## Running the sample
 
-Outline step-by-step instructions to execute the sample and see its output. Include steps for executing the sample from the IDE, starting specific services in the Azure portal or anything related to the overall launch of the code.
+1. To start the sample application, run `npm start`.
+2. Next, open a browser to [http://localhost:4200](http://localhost:4200).
 
 ## Key concepts
 
-Provide users with more context on the tools and services used in the sample. Explain some of the code that is being used and how services interact with each other.
+This sample demonstrates the follow MSAL Angular concepts:
+
+* Configuration
+* Login
+* Logout
+* Protecting a route
+* Acquire an access token and attaching it to http calls
 
 ## Contributing
 
-This project welcomes contributions and suggestions.  Most contributions require you to agree to a
-Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us
-the rights to use your contribution. For details, visit https://cla.opensource.microsoft.com.
+If you'd like o contribute to this sample, see [CONTRIBUTING.MD](./CONTRIBUTING.md).
 
-When you submit a pull request, a CLA bot will automatically determine whether you need to provide
-a CLA and decorate the PR appropriately (e.g., status check, comment). Simply follow the instructions
-provided by the bot. You will only need to do this once across all repos using our CLA.
+## Code of Conduct
 
 This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
 For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or

```

Comment:
---
page_type: sample
languages:
- javascript
- nodejs
products:
- azure-active-directory
description: "A simple Angular single page application calling the Microsoft Graph using MSAL Angular (Azure AD V2 endpoint)."
urlFragment: "active-directory-javascript-singlepageapp-angular"
---

# MSAL Angular Sample Application

A simple Angular single-page application which demonstrates how to configure [MSAL Angular](https://www.npmjs.com/package/@azure/msal-angular) to login, logout, protect a route, and acquire an access token for a protected resource such as the Microsoft Graph.

**Note:** This sample's structure was generated with the [Angular CLI](https://cli.angular.io/).

## Contents

| File/folder       | Description                                |
|-------------------|--------------------------------------------|
| `e2e`             | End-to-end test files.                     |
| `src`             | Sample source code.                        |
| `.editorconfig`   | Defines editor config settings.            |
| `.gitignore`      | Define what to ignore at commit time.      |
| `angular.json`    | Angular configuration file.                |
| `browserslist`    | BrowsersList configuration file.           |
| `CHANGELOG.md`    | List of changes to the sample.             |
| `CODE_OF_CONDUCT.md` | Code of Conduct information.            |
| `CONTRIBUTING.md` | Guidelines for contributing to the sample. |
| `karma.conf.js  ` | Configuration for the karma test runner.   |
| `LICENSE`         | The license for the sample.                |
| `NG_README.md`    | README auto-generated by the Angular CLI.  |
| `package-lock.json` | Lockfile for npm. |
| `package.json`    | Package manifest for npm.                   |
| `README.md`       | This README file.                          |
| `SECURITY.md`     | Security disclosures.                      |
| `tsconfig.*.json` | TypeScript configuration files.            |
| `tslint.json`     | TS Lint configuration files.               |

## Prerequisites

[Node](https://nodejs.org/en/) must be installed to run this sample.

## Setup

1. [Register a new application](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration) in the [Azure Portal](https://portal.azure.com). Ensure that the application is enabled for the [implicit flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow).
2. Open the [src/app/app.module.ts](./src/app/app.module.ts) file and provide the required configuration values (inside `MsalModule.forRoot`).
3. On the command line, navigate to the root of the repository, and run `npm install` to install the project dependencies via npm.

## Running the sample

1. To start the sample application, run `npm start`.
2. Next, open a browser to [http://localhost:4200](http://localhost:4200).

## Key concepts

This sample demonstrates the following MSAL Angular concepts:

* Configuration
* Login
* Logout
* Protecting a route
* Acquiring an access token and attaching it to http calls

## Contributing

If you'd like to contribute to this sample, see [CONTRIBUTING.MD](./CONTRIBUTING.md).

## Code of Conduct

This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Reply:
Thanks!

For the implicit flow, it already links to [this page](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration), which has instructions for how to enable the implicit flow, which I think is the primary thing customers would need to know. Cool if I include the link you added as a separate link?

Reply:
Sure, that would work as well. I'd say a small direct link to implicit flow is unobtrusive and provides ease of access (even though not every customer will be interested in details, some might want a refresher) -but anyways, I'll defer to your on this!

Reply:
Oh I see. Can I get access to that tool as well? I could use it with the sample I'm working on

Reply:
Re: links for implicit flow, I missed that you made "Register a new application" a link, so I'm cool with that.